src/formatter.ts

```diff
@@ -19,6 +19,16 @@ import { isNote, isStaveNote } from './typeguard';
 import { defined, log, midLine, RuntimeError } from './util';
 import { Voice } from './voice';
 
+// Default softmax.  Can be set on measure level with FormatterOptions
+// or globally by changing the default
+export let softmaxDefault = 100;
```

Comment:
I think that this declaration should be in `Tables` as a static member. (I would probaby go for softmaxFactor rather than softmaxDefault if we agree that it makes no sense to have different values on different voices)

Reply:
I think softmaxDefault and the setter has to be in the same file.  And I didn't want to put it in Tables.js because that is not accessible from outside of vexflow.

I could put both the variable and the setters in flow - will that work?

Reply:
My proposal is to do it like `setMusicFont` / `getMusicFont`

`static SOFTMAX_DEFAULT = 100` in `Tables` or `SOFMAX_FACTOR` if we agree on my oppnion in the discussion above.

Reply:
> My proposal is to do it like setMusicFont / getMusicFont


It seems like this should work, but I'm getting this strange error.  Any ideas why?  I notice that there is nowhere in the test files or the code base where 'getMusicFont' is actually used, I wonder if that would product a similar error.  (it is late here so maybe I am overlooking something obvious.  Surprising that it compiles fine)

Also, this makes it so setting softmax for the measure occurs in a different module that setting softmax globally, which is a little odd.

anyway, here is the error that introduces.
```
barnote.ts:79 Uncaught ReferenceError: Cannot access 'Beam' before initialization
    at Module.Beam (barnote.ts:79:1)
    at Object../src/flow.ts (flow.ts:247:13)
    at __webpack_require__ (bootstrap:19:1)
    at Object../src/formatter.ts (textfonts.ts:52:1)
    at __webpack_require__ (bootstrap:19:1)
    at Object../src/tuplet.ts (tuning.ts:70:1)
    at __webpack_require__ (bootstrap:19:1)
    at Object../src/beam.ts (barnote.ts:79:1)
    at __webpack_require__ (bootstrap:19:1)
    at Object../src/index.ts (gracetabnote.ts:28:1)
Beam @ barnote.ts:79
./src/flow.ts @ flow.ts:247
__webpack_require__ @ bootstrap:19
./src/formatter.ts @ textfonts.ts:52
__webpack_require__ @ bootstrap:19
./src/tuplet.ts @ tuning.ts:70
__webpack_require__ @ bootstrap:19
./src/beam.ts @ barnote.ts:79
__webpack_require__ @ bootstrap:19
./src/index.ts @ gracetabnote.ts:28
__webpack_require__ @ bootstrap:19
(anonymous) @ make namespace object:7
(anonymous) @ vexflow-debug-with-tests.ts:19
(anonymous) @ vexflow-debug-with-tests.ts:19
webpackUniversalModuleDefinition @ universalModuleDefinition:9
(anonymous) @ universalModuleDefinition:10
23:07:47.525 flow.html?module=Bach%20Demo:139 Uncaught ReferenceError: Vex is not defined
    at flow.html?module=Bach%20Demo:139:18
(anonymous) @ flow.html?module=Bach%20Demo:139
load (async)
(anonymous) @ flow.html?module=Bach%20Demo:38
loadScript @ flow.html?module=Bach%20Demo:36
loadVexFlow @ flow.html?module=Bach%20Demo:125
(anonymous) @ flow.html?module=Bach%20Demo:137

```

Reply:
Whenever I see "Cannot access 'XXXX' before initialization", I assume it's a circular dependency.

It might compile correctly, but when running, it means that class A imported class B, which during initialization requires class A (it might be a few dependencies down, but eventually it requires class A).

Unfortunately, class A hasn't been fully initialized yet since it was still in the middle of importing class B, so some of its static properties are still undefined.

Sorry I haven't been following this PR closely. Are we trying to add a static property to Tables to customize the softmax value?

If all external accesses are through the Flow class, I don't see why we can't keep the actual property inside Tables.


FYI: I implemented getMusicFont(). I thought I did use it somewhere (maybe in the demos but not in the unit tests). If I skipped it, it's my fault, haha.

Let me know if you need help debugging this (esp if you think it's a circular dependency).

Reply:
It is definitely a circular dependency.  flow requires beam requires tuplet which requires formatter.  So we can't add a dependency from formatter to flow without further changes.  I am going back to softmax default as global variable in formatter.  Not everything needs to be part of a class.